### PR TITLE
fix bazel incompatibilities in proto

### PR DESCRIPTION
**Pull request by johnynek**
just the proto part of #452 to address #430

### CODE REVIEW

File: scala_proto/scala_proto.bzl [hunk 1]
```diff
@@ -319,38 +319,39 @@ def _colon_paths(data):
...
             proto_deps.append(target)
-            acc_imports += target.proto.transitive_sources
+            acc_imports.append(target.proto.transitive_sources)
             #inline this if after 0.12.0 is the oldest supported version
             if hasattr(target.proto, 'transitive_proto_path'):
               transitive_proto_paths += target.proto.transitive_proto_path
         else:
             jvm_deps.append(target)
 
+    acc_imports = depset(transitive = acc_imports)
```
Comment (ittaiz) [inline scala_proto/scala_proto.bzl]: Can we move changing acc_imports to a depset after 44? 
In it you’re changing it back to a list which seems redundant

File: scala_proto/scala_proto.bzl [hunk 2]
```diff
@@ -319,38 +319,39 @@ def _colon_paths(data):
...
-    acc_imports = depset()
+    acc_imports = []
     transitive_proto_paths = depset()
 
     proto_deps, jvm_deps = [], []
     for target in ctx.attr.deps:
         if hasattr(target, 'proto'):
             proto_deps.append(target)
-            acc_imports += target.proto.transitive_sources
+            acc_imports.append(target.proto.transitive_sources)
```
Comment (ittaiz) [inline scala_proto/scala_proto.bzl]: Isn’t transitive_sources a depset? If so isn’t it better to accumulate them in depsets?

File: scala_proto/scala_proto.bzl (hunk 1 above)
Comment (johnynek) [reply scala_proto/scala_proto.bzl]: so, note that `acc_imports` as the arg to transitive is a `List[Depset]` not a `List[File]` or something, so when we do `depset(transitive = acc_imports).to_list()` converts a `List[Depset]` to a `List[File]`.

File: scala_proto/scala_proto.bzl (hunk 2 above)
Comment (johnynek) [reply scala_proto/scala_proto.bzl]: there is no way to accumulate depsets that is not deprecated. You accumulate a list and then make a `depset(transitive = your_list_here)`

Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: @laurentlb is this correct? Why can’t we accumulate depsets? Turning depsets to lists so we accumulate them as depsets sounds counter intuitive

File: scala_proto/scala_proto.bzl (hunk 1 above)
Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: 😱 that's so implicit JavaConversions feels explicit 😉

File: scala_proto/scala_proto.bzl (hunk 2 above)
Comment (laurentlb) [reply scala_proto/scala_proto.bzl]: Yes, it's correct (https://docs.bazel.build/versions/master/skylark/depsets.html).
* depset is immutable, so the `+=` operator was misleading
* we can create only one `depset`, with the correct structure, and it's much better for performance (adding depset inside a loop could lead to `O(n^2)` complexity
* it's more explicit: you see that the depset has n transitive depsets at the same level (which corresponds to the internal representation)

Depsets are not turned into lists. We just collect all the depsets first, and then merge them all at the same time.

Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: Can we pass a depset to the “transitive” argument of the constructor?
What Oscar did is convert them to lists for this ctor which sounds wrong

Comment (johnynek) [reply scala_proto/scala_proto.bzl]: I didn’t convert them to lists. I added them to a list. The argument to transitive is a list of Depsets. I constructed that list so I could pass the list of depsets to transitive.

Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: I see, my bad.
Thanks for the clarification

**Pull request merged**